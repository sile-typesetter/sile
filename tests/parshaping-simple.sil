\begin[papersize=a7]{document}
\nofolios
\script[src=packages/lorem]
\begin{script}

local testshape = {
  { 50, 30, nil },
  { 50, 60, nil },
  { nil, 120, nil },
  { nil, 30, 40 },
  { nil, nil, nil },
  { 50, nil, 50 },
}

SILE.settings.set("linebreak.tolerance", 2000)
SILE.settings.set("document.parindent", 0)
SILE.settings.set("linebreak.parShape", true)

SILE.linebreak.parShape = function (self, line)
  local shape = testshape[line]
	if shape then
		return shape[1], shape[2], shape[3]
	else
		return nil, nil, nil
	end
end

SILE.call("lorem", { words = 20 })
SILE.call("par")
SILE.call("lorem", { words = 30 })

\end{script}
\end{document}
