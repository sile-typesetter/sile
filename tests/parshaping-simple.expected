Set paper size 	209.7637818	297.6377985
Begin page
Mx 	60.4882
My 	22.4502
Set font 	Gentium Plus;10;400;;normal;;LTR
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	86.7473
T	76 83 w=8.0078	(ip)
Mx 	94.7551
T	16 w=3.3691	(-)
Mx 	60.4882
My 	34.4502
T	86 88 80 w=17.1924	(sum)
Mx 	83.1127
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	110.4735
T	86 76 87 w=10.0146	(sit)
Mx 	10.4882
My 	46.4502
T	68 80 72 87 w=20.6836	(amet)
Mx 	33.0803
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	79.1490
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	124.0556
T	72 79 76 87 85 w=17.4414	(elitr)
Mx 	129.2756
My 	58.4502
T	86 72 71 w=13.6816	(sed)
Mx 	144.8658
T	71 76 68 80 w=20.5322	(diam)
Mx 	10.4882
My 	70.4502
T	81 82 81 88 80 92 w=34.3262	(nonumy)
Mx 	49.7095
T	72 76 85 80 82 71 w=29.5508	(eirmod)
Mx 	84.1555
T	87 72 80 83 82 85 w=30.3809	(tempor)
Mx 	119.4315
T	76 81 89 76 71 88 81 87 w=35.3271	(invidunt)
Mx 	159.6539
T	88 87 w=8.7402	(ut)
Mx 	173.2893
T	79 68 69 82 85 72 w=25.9863	(labore)
Mx 	60.4882
My 	82.4502
T	72 87 w=8.0615	(et)
Mx 	72.9469
T	71 82 79 82 85 72 w=26.5479	(dolore)
Mx 	103.8920
T	80 68 74 81 68 w=27.6074	(magna)
Mx 	135.8967
T	68 79 76 w=10.0098	(ali)
Mx 	145.9065
T	16 w=3.3691	(-)
Mx 	10.4882
My 	94.4502
T	84 88 92 68 80 w=28.0029	(quyam)
Mx 	60.4882
My 	106.4502
T	79 82 85 72 80 w=24.3506	(lorem)
Mx 	86.7473
T	76 83 w=8.0078	(ip)
Mx 	94.7551
T	16 w=3.3691	(-)
Mx 	60.4882
My 	118.4502
T	86 88 80 w=17.1924	(sum)
Mx 	83.1127
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	110.4735
T	86 76 87 w=10.0146	(sit)
Mx 	10.4882
My 	130.4502
T	68 80 72 87 w=20.6836	(amet)
Mx 	33.0803
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	79.1490
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	124.0556
T	72 79 76 87 85 w=17.4414	(elitr)
Mx 	129.2756
My 	142.4502
T	86 72 71 w=13.6816	(sed)
Mx 	144.8658
T	71 76 68 80 w=20.5322	(diam)
Mx 	10.4882
My 	154.4502
T	81 82 81 88 80 92 w=34.3262	(nonumy)
Mx 	49.7095
T	72 76 85 80 82 71 w=29.5508	(eirmod)
Mx 	84.1555
T	87 72 80 83 82 85 w=30.3809	(tempor)
Mx 	119.4315
T	76 81 89 76 71 88 81 87 w=35.3271	(invidunt)
Mx 	159.6539
T	88 87 w=8.7402	(ut)
Mx 	173.2893
T	79 68 69 82 85 72 w=25.9863	(labore)
Mx 	60.4882
My 	166.4502
T	72 87 w=8.0615	(et)
Mx 	72.9469
T	71 82 79 82 85 72 w=26.5479	(dolore)
Mx 	103.8920
T	80 68 74 81 68 w=27.6074	(magna)
Mx 	135.8967
T	68 79 76 w=10.0098	(ali)
Mx 	145.9065
T	16 w=3.3691	(-)
Mx 	10.4882
My 	178.4502
T	84 88 92 68 80 w=28.0029	(quyam)
Mx 	41.6434
T	72 85 68 87 w=16.6113	(erat)
Mx 	61.4070
T	86 72 71 w=13.6816	(sed)
Mx 	78.2409
T	71 76 68 80 w=20.5322	(diam)
Mx 	101.9253
T	89 82 79 88 83 87 88 68 w=36.5967	(voluptua)
Mx 	141.6743
T	68 87 w=8.0322	(at)
Mx 	152.8588
T	89 72 85 82 w=18.5400	(vero)
Mx 	174.5511
T	72 82 86 w=13.5107	(eos)
Mx 	191.2141
T	72 87 w=8.0615	(et)
Mx 	10.4882
My 	190.4502
T	68 70 70 88 86 68 80 w=35.1123	(accusam)
Mx 	48.2588
T	72 87 w=8.0615	(et)
End page
Finish
